FindTown/FindTown/Presentation/HomeScene/Home/Sheets/FilterBottomSheetViewModel.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  FilterBottomSheetViewModel.swift
+//  FindTown
+//
+//  Created by 김성훈 on 2023/01/16.
+//
+
+import UIKit
+import FindTownCore
+
+import RxSwift
+import RxRelay
+
+protocol FilterBottomSheetViewModelDelegate {
+    func dismiss(_ tempModel: TempFilterModel)
+}
+
+protocol FilterBottomSheetViewModelType {
+    func dismiss(_ tempModel: TempFilterModel)
+}
+
+// 임시
+struct TempFilterModel {
+    var infra: String = ""
+    var traffic: [String] = []
+}
+
+final class FilterBottomSheetViewModel: BaseViewModel {
+    
+    struct Input {
+        let infra = PublishSubject<String>()
+        let traffic = PublishSubject<[String]>()
+        let completeButtonTrigger = PublishSubject<Void>()
+    }
+    
+    struct Output {
+        let trafficDataSource = BehaviorSubject<[Traffic]>(value: [])
+        let buttonsSelected = BehaviorRelay<Bool>(value: false)
+    }
+    
+    let input = Input()
+    let output = Output()
+    let delegate: FilterBottomSheetViewModelDelegate
+    private var tempFilterModel: TempFilterModel?
+    
+    init(delegate: FilterBottomSheetViewModelDelegate) {
+        self.delegate = delegate
+        self.tempFilterModel = TempFilterModel()
+        
+        super.init()
+        self.bind()
+    }
+    
+    func bind() {
+        
+        self.output.trafficDataSource.onNext(Traffic.allCases)
+
+        self.input.completeButtonTrigger
+            .bind { [weak self] in
+                self?.dismiss(self?.tempFilterModel ?? TempFilterModel.init())
+            }
+            .disposed(by: disposeBag)
+        
+        self.input.infra
+            .map { [weak self] infra in
+                self?.tempFilterModel?.infra = infra
+                return infra != ""
+            }
+            .bind { [weak self] in
+                self?.output.buttonsSelected.accept($0)
+            }
+            .disposed(by: disposeBag)
```

Comment:
```suggestion
         self.input.infra
            .map { return ($0, $0 != "")}
            .bind { [weak self] in
                self?.tempFilterModel?.infra = $0.0
                self?.output.buttonsSelected.accept($0.1)
            }
            .disposed(by: disposeBag)
```
map에서 속성을 설정하는게 조금.. 어색한 것 같아서요! map에서 튜플로 넘겨주면 어떨까 싶어서 남겨봅니당..

Reply:
오우.. 생각지도 못한 방법이네요.. 너무 좋습니다!